packages/nimble-components/src/table-column/base/index.ts

```diff
@@ -163,4 +199,26 @@ export abstract class TableColumn<
     protected internalPixelWidthChanged(): void {
         this.currentPixelWidth = this.internalPixelWidth;
     }
+
+    protected groupHeaderViewTagChanged(): void {
+        this._groupHeaderViewTemplate = this.createGroupHeaderViewTemplate();
+        Observable.notify(this, 'internalGroupHeaderViewTemplate');
+    }
+
+    private createGroupHeaderViewTemplate():
+    | ViewTemplate<TableGroupRow>
+    | undefined {
+        if (!this.groupHeaderViewTag) {
+            return undefined;
+        }
+
+        return html<TableGroupRow>`
+            <${this.groupHeaderViewTag}
+                :groupHeaderValue="${x => x.groupRowValue}"
+                :columnConfig="${x => x.groupColumn?.columnConfig}"
+                class="group-header-value"
+                >
+            </${this.groupHeaderViewTag}>
+        `;
+    }
```

Comment:
Maybe I'm missing something but is there a reason for all this additional complexity with custom notifiers and observers?

Was thinking it would look similar to the pattern being done for the cell-view: https://github.com/ni/nimble/blob/806a8a52913042fc0e9d54f60aea88aaec46ffad/packages/nimble-components/src/table-column/base/index.ts#L80-L90

Another benefit of the cell view pattern is that the contents of the template itself don't have to live in this file:
https://github.com/ni/nimble/blob/806a8a52913042fc0e9d54f60aea88aaec46ffad/packages/nimble-components/src/table-column/base/index.ts#L154-L159

Reply:
I could be over-thinking things, but I wanted to create a read-only property for the `groupHeaderViewTemplate` (even with it being `internal`), but still have it notify when its backing field changed. I don't mind removing this slight additional complexity if you feel that's better.

Definitely will move the `createGroupHeaderViewTemplate` function to a new `template.ts` file under the `group-header-view` folder in `table\components` (to match what the `cellView` does).

Reply:
Went ahead and just did as @rajsite suggested, as there was no reason to diverge from how we've been defining all of the other `internal` properties.